File: TechModule/Programming Fundamentals/04.Arrays - Exercises/10.EqualSums/EqualSums.cs
using System;
using System.Linq;

class EqualSums
{
    static void Main()
    {
        int[] array = Console.ReadLine().Split(new char[] { ' ' }, StringSplitOptions.RemoveEmptyEntries).Select(int.Parse).ToArray();

        bool exists = false;
        for (int i = 0; i < array.Length; i++)
        {
            int leftSum = 0;
            int rightSum = 0;
            for (int j = 0; j < i; j++)
            {
                leftSum += array[j];
            }

            for (int j = i + 1; j < array.Length; j++)
            {
                rightSum += array[j];
            }

            if (leftSum == rightSum)
            {
                Console.WriteLine(i);
                exists = true;
                break;
            }
        }

        if (!exists)
        {
            Console.WriteLine("no");
        }
    }
}

File: Java-DB-Fundamentals/DatabasesFrameworks-Hibernate&SpringData/01.JavaBasicSyntax/src/P11/EqualSums.java
package P11;

import java.util.Arrays;
import java.util.Scanner;

public class EqualSums {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int[] arr = Arrays.stream(scan.nextLine().split("\\s")).mapToInt(a->Integer.parseInt(a)).toArray();
        int leftSum = 0;
        int rightSum = 0;
        boolean exists = false;
        for (int i = 0; i <arr.length ; i++) {
            leftSum = 0;
            rightSum = 0;
            for (int j = 0; j <i ; j++) {
                leftSum +=arr[j];
            }
            for (int j = i+1; j < arr.length; j++) {
                rightSum += arr[j];
            }
            if (leftSum == rightSum){
                System.out.println(i);
                exists = true;
                break;
            }
        }
        if (!exists){
            System.out.println("no");
        }
    }
}

